---
title_full: "faers: An R Package for Seamlessly Bridging the FAERS Database to R and Delivering Unified Pharmacovigilance Workflows"
title_short: Bioinformatics Rmd Template

# One of the following below. See https://academic.oup.com/bioinformatics/pages/instructions_for_authors#Types%20of%20Manuscript   
# one of `Original Paper`, `Discovery Note`, `Applications Note`, `Review`, `Letter to the Editor`, `Conference Paper`    
manuscript_type: Application Note

# change accordingly
copyrightyear: '`r format(Sys.Date(), "%Y")`'
pubyear: '`r format(Sys.Date(), "%Y")`'

# One of the following below. See https://academic.oup.com/bioinformatics/pages/instructions_for_authors#Detailed%20Scope%20Guidelines   
# Genome analysis
# Sequence analysis
# Phylogenetics
# Structural bioinformatics
# Gene expression
# Genetic and population analysis
# Systems biology
# Data and text mining
# Databases and ontologies
# Bioimage informatics
subject_section:  Genome analysis

access_date:
  - day: DD
    month: MM
    year: YY
      
author:
  - name: Shixiang Wang
    affiliation: "1"
  - name: Yun Peng
    affiliation: "2"
  - name: Zhangyu Wang
    affiliation: "1"
      
address:
  - code: 1
    address:  Central South University
  - code: 2
    address:  aaa
 
# Full email address to be given, preferably an institution email address.  
corresponding_author_text: "To whom correspondence should be addressed. E-mail: qq"
corresponding_author_email: "212&#64;qq.com"
  
motivation:  |
 The FDA Adverse Event Reporting System (FAERS) serves as a critical database for monitoring post-marketing drug safety.However, the primary focus on safety signal detection within FAERS has left a significant gap in integrating pharmacovigilance analysis with genetic tools.Thus, we aim to effectively utilize FAERS data to bridge pharmacovigilance with genetic analysis, thereby enhancing the precision of safety assessments and facilitating the development of personalized medicine approaches.

  
results: |
 We developed the R package faers, which seamlessly connects the FAERS database to the R programming language and provides a unified approach for the seamless execution of pharmacovigilance analysis and the integration of genetic tools in R.

availability: |
 faers is available on CRAN and on GitHub at (https&#58;&#47;&#47;github.com/Yunuuuu/faers).

supplementary_info: |
  Supplementary data are available at Bioinformatics Online.

output:
  rticles::bioinformatics_article:
    citation_package: natbib
bibliography: bibliography.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(eval = FALSE)
devtools::load_all("/home/wzy/projects/test/faers")
library(faers)
faers_meta()
data1<-faers(2021,"q1",dir = "~/projects/test/data",compress_dir ="~/projects/test/data")
data_stand <- faers_standardize(data1, "~/projects/test/meddra")
data_stand <- faers_dedup(data_stand)
```

```{r analysis, include=FALSE}
insulin_names <- "insulin"
insulin_pattern <- paste(insulin_names, collapse = "|")
fda_insulin <- fda_drugs()[
    grepl(insulin_pattern, ActiveIngredient, ignore.case = TRUE)
]
```

Introduction
============

The FDA Adverse Event Reporting System (FAERS) is the core database for post-marketing safety monitoring of all approved drugs and therapeutic biologics by the FDA. As required by regulations, it consolidates mandatory adverse event reports submitted by pharmaceutical companies, as well as voluntary first-hand information reported by individuals and healthcare professionals. The data, which has been collected since January 2004 and is continuously updated on a quarterly basis, includes eight types of documents: demographic and administrative information, detailed drug information, indications for use, report sources, drug start and end dates, patient outcomes, reports of therapeutic failure, and adverse events. Its core value is reflected in key areas such as early safety signal detection, risk assessment, and drug labeling revisions, providing direct evidence for the FDA to formulate drug safety policies and risk management measures.

Currently, FAERS research has formed a tripartite landscape of "official—open-source—R ecosystem": The official path still requires researchers to download quarterly files, deduplicate and clean them, recode with MedDRA, and then manually invoke algorithms. Although the FAERS Public Dashboard provides zero-code visualization and download, it lacks a batch analysis interface. In the open-source platform, OpenVigil 2.1, by pre-hosting cleaned data from 2004 to 2022, supports one-click output of PRR/ROR after online setting of drug-event-time window, but its cross-platform reproducibility is limited. AERSMine, leveraging Spark to integrate 2.1 million reports, offers interactive differential efficacy forest plots, yet still cannot directly connect to the Bioconductor genetics pipeline. DiAna_package encapsulates downloading, cleaning, standardization, signaling, and genetics integration into a single R command, being the first to incorporate population genotypes into the signal model, but it leaves gaps in environment locking and full-field integrity, which provides an entry point for the proposed unified faers framework in this paper.

To systematically address the shortcomings of existing tools in terms of reproducibility, integration into the R ecosystem, and in-depth mining of genetic information, we developed the faers package. This package connects the FAERS database to R through a unified pipeline, enabling one-click data download, cleaning, standardization, and signal detection. It also seamlessly integrates with the Bioconductor genetics analysis framework, supporting high-throughput association analysis of population-differentiated polymorphisms and drug pathways, thereby providing reproducible and scalable decision support for precision medicine.

Methods
============

The workflow for pharmacovigilance analysis using the faers package mainly consists of the following four steps: (i) inspecting FAERS metadata; (ii) downloading and parsing the quarterly FAERS data files; (iii) standardization and deduplication; (iv) conducting pharmacovigilance analysis.

Check metadata of FAERS
-------------

Details for Method 1. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Details for Method 1. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Details for Method 1. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Details for Method 1. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Method 2
-------------

Details for Method 2. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Details for Method 2. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Details for Method 2. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Details for Method 2. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Discussion
=============

Discussion of results. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Discussion of results. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Discussion of results. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Discussion of results. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Conclusion
=============

Anything else? Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Anything else? Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Anything else? Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Anything else? Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum. Introduce your topic. Lorem ipsum ad nauseum.

Acknowledgements {.unnumbered}
==============

These should be included at the end of the text and not in footnotes. Please ensure you acknowledge all sources of funding, see funding section below.

Details of all funding sources for the work in question should be given in a separate section entitled 'Funding'. This should appear before the 'Acknowledgements' section.

Funding {.unnumbered}
==============
The following rules should be followed:

- The sentence should begin: ‘This work was supported by …’ -
- The full official funding agency name should be given, i.e. ‘National Institutes of Health’, not ‘NIH’ (full RIN-approved list of UK funding agencies)
- Grant numbers should be given in brackets as follows: ‘[grant number xxxx]’
- Multiple grant numbers should be separated by a comma as follows: ‘[grant numbers xxxx, yyyy]’
- Agencies should be separated by a semi-colon (plus ‘and’ before the last funding agency) 
- Where individuals need to be specified for certain sources of funding the following text should be added after the relevant agency or grant number 'to [author initials]'.

An example is given here: ‘This work was supported by the National Institutes of Health [AA123456 to C.S., BB765432 to M.H.]; and the Alcohol & Education Research Council [hfygr667789].’

Oxford Journals will deposit all NIH-funded articles in PubMed Central. See Depositing articles in repositories – information for authors for details. Authors must ensure that manuscripts are clearly indicated as NIH-funded using the guidelines above.

